### PR TITLE
Adds and expands on the descriptions of tank hardpoints & magazins

### PR DESCRIPTION
**Pull request by TheManWithNoHands**
# About the pull request

* Adds descriptions in the vehicle vendor to tank hardpoint weapon options. Giving a small description to how the guns work.
* Gives the magazins and hardpoints item descriptions a more flavorfull description, being that most where literally placeholders.

# Explain why it's good for the game

Currently you dont really know what the gun does you select for the tank. Wich is kinda shit.
Your lucky, and finally get to play tank crew for the first time... and you have no clue what any of the guns do. This should help with that.
Being that, you know, you can now read what the gun does your commiting to.

Also, a lot of descriptions where 100% placeholders of the items. The tank is really in dire need of some updates.
So i gave most of them description with some more flavor to them then just "Its a tank primary".

Also, im kinda unsure what type of changelog this applys to.
# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:TheManWithNoHands
add: Added vendor description to: AC3-E autocanon, LTAA-AP Minigun, DRG-N Flamer, LTB canon, LZR-N Flamer, TOW launcher, M56cupola, and M92T Grenade Launcher.
add: Expanded the description of: LTAA-AP Minigun, TOW launcher, M56cupola, M92T Grenade Launcher and the magazins of the AC3-E autocanon, LTAA-AP Minigun, DRG-N Flamer, LZR-N Flamer, and M92T Grenade Launcher. As well as for the snowplow hardpoint and heavy and ligth treads.
/:cl:

### CODE REVIEW

File: code/game/machinery/vending/vendor_types/crew/vehicle_crew.dm [hunk 1]
```diff
@@ -353,6 +353,7 @@ GLOBAL_LIST_INIT(cm_vending_clothing_vehicle_crew, list(
 
 //Not essentials sets but fuck it the code's here
 /obj/effect/essentials_set/tank/ltb
+	desc = "A giant canon firing explosive 86mm shells. You be lucky if this even leaves dust of whatever you hit with it."
```
Comment (harryob) [inline code/game/machinery/vending/vendor_types/crew/vehicle_crew.dm:356]:
```suggestion
	desc = "A giant canon firing explosive 86mm shells. You'd be lucky if this even leaves the dust of whatever you hit with it."
```

File: code/game/machinery/vending/vendor_types/crew/vehicle_crew.dm [hunk 2]
```diff
@@ -363,20 +364,23 @@ GLOBAL_LIST_INIT(cm_vending_clothing_vehicle_crew, list(
 	)
 
 /obj/effect/essentials_set/tank/gatling
+	desc = "A primary LTAA Minigun utelizing AP ammo for tanks. The barrel spins up as it is fired, improving its fire rate and accuracy dramaticly. Capable of shreeding apart even the thickest walls in seconds."
```
Comment (harryob) [inline code/game/machinery/vending/vendor_types/crew/vehicle_crew.dm:367]:
```suggestion
	desc = "A primary LTAA Minigun utilizing AP ammo for tanks. The barrel spins up as it is fired, improving its fire rate and accuracy dramatically. Capable of shredding apart even the thickest walls in seconds."
```

File: code/game/machinery/vending/vendor_types/crew/vehicle_crew.dm [hunk 3]
```diff
@@ -363,20 +364,23 @@ GLOBAL_LIST_INIT(cm_vending_clothing_vehicle_crew, list(
...
 /obj/effect/essentials_set/tank/dragonflamer
+	desc = "A heavy flamer that spews out high-combustion napalm in a wide radius. The fuel burns intensely and quickly, which allows for it to be used offensively by armoured vehicles."
 	spawned_gear_list = list(
 		/obj/item/hardpoint/primary/flamer,
 		/obj/item/ammo_magazine/hardpoint/primary_flamer,
 		/obj/item/ammo_magazine/hardpoint/primary_flamer,
 	)
 
 /obj/effect/essentials_set/tank/autocannon
+	desc = "An Automatic Canon for tanks. Capable of firing precise even at long ranges. Loads 20mm explosive shells."
```
Comment (harryob) [inline code/game/machinery/vending/vendor_types/crew/vehicle_crew.dm:383]:
```suggestion
	desc = "An automatic cannon for tanks, capable of firing precisely even at long ranges. Loads 20mm explosive shells."
```

**Review by harryob — CHANGES_REQUESTED** [review]

**Comment by TheManWithNoHands** [comment]
German grammar rules be rooted deep

File: code/game/machinery/vending/vendor_types/crew/vehicle_crew.dm [hunk 4]
```diff
@@ -386,25 +390,29 @@ GLOBAL_LIST_INIT(cm_vending_clothing_vehicle_crew, list(
 	)
 
 /obj/effect/essentials_set/tank/tankflamer
+	desc = "A small LZR-N Flamer Unit. Essentially a modified version of your bog standart flamer."
```
Comment (harryob) [inline code/game/machinery/vending/vendor_types/crew/vehicle_crew.dm:393]:
```suggestion
	desc = "A small LZR-N Flamer Unit - a modified version of your bog standard flamer."
```

File: code/game/machinery/vending/vendor_types/crew/vehicle_crew.dm [hunk 5]
```diff
@@ -386,25 +390,29 @@ GLOBAL_LIST_INIT(cm_vending_clothing_vehicle_crew, list(
...
 	spawned_gear_list = list(
 		/obj/item/hardpoint/secondary/small_flamer,
 		/obj/item/ammo_magazine/hardpoint/secondary_flamer,
 	)
 
 /obj/effect/essentials_set/tank/tow
+	desc = "A quint rocket launcher capable of firing four rockets in quick succession."
```
Comment (harryob) [inline code/game/machinery/vending/vendor_types/crew/vehicle_crew.dm:400]:
```suggestion
	desc = "A quad rocket launcher capable of firing four rockets in quick succession."
```

File: code/game/machinery/vending/vendor_types/crew/vehicle_crew.dm [hunk 6]
```diff
@@ -386,25 +390,29 @@ GLOBAL_LIST_INIT(cm_vending_clothing_vehicle_crew, list(
...
 	spawned_gear_list = list(
 		/obj/item/hardpoint/secondary/towlauncher,
 		/obj/item/ammo_magazine/hardpoint/towlauncher,
 		/obj/item/ammo_magazine/hardpoint/towlauncher,
 	)
 
 /obj/effect/essentials_set/tank/m56cupola
+	desc = "A permantly fixed M56D, firing standart issue 10x28mm rounds."
```
Comment (harryob) [inline code/game/machinery/vending/vendor_types/crew/vehicle_crew.dm:408]:
```suggestion
	desc = "A permanently fixed M56D, firing standard issue 10x28mm rounds."
```

File: code/game/machinery/vending/vendor_types/crew/vehicle_crew.dm [hunk 7]
```diff
@@ -386,25 +390,29 @@ GLOBAL_LIST_INIT(cm_vending_clothing_vehicle_crew, list(
...
 	spawned_gear_list = list(
 		/obj/item/hardpoint/secondary/m56cupola,
 		/obj/item/ammo_magazine/hardpoint/m56_cupola,
 	)
 
 /obj/effect/essentials_set/tank/tankgl
+	desc = "A magazin feed grenade launcher capable of holding 10 grenades. This model loads M40 grenades."
```
Comment (harryob) [inline code/game/machinery/vending/vendor_types/crew/vehicle_crew.dm:415]:
```suggestion
	desc = "A magazine feed grenade launcher capable of holding 10 grenades. This model loads M40 grenades."
```

File: code/modules/vehicles/hardpoints/armor/snowplow.dm [hunk 8]
```diff
@@ -1,6 +1,6 @@
 /obj/item/hardpoint/armor/snowplow
 	name = "\improper Snowplow"
-	desc = "Clears a path in the snow for friendlies"
+	desc = "Clears a path in the snow for friendlies. It doesnt seem to have much use beyond that."
```
Comment (harryob) [inline code/modules/vehicles/hardpoints/armor/snowplow.dm:3]:
```suggestion
	desc = "Clears a path in the snow for friendlies. It doesn't seem to have much use beyond that."
```

File: code/modules/vehicles/hardpoints/hardpoint_ammo/autocannon_ammo.dm [hunk 9]
```diff
@@ -1,6 +1,6 @@
 /obj/item/ammo_magazine/hardpoint/ace_autocannon
-	name = "Tank Autocannon Magazine"
-	desc = "A primary armament autocannon magazine"
+	name = "AC3-E Autocannon Magazine"
+	desc = "A 40 round magazin holding 20mm shells for the AC3-E autocanon"
```
Comment (harryob) [inline code/modules/vehicles/hardpoints/hardpoint_ammo/autocannon_ammo.dm:3]:
```suggestion
	desc = "A 40 round magazine holding 20mm shells for the AC3-E autocannon."
```

File: code/modules/vehicles/hardpoints/hardpoint_ammo/gl_ammo.dm [hunk 10]
```diff
@@ -1,6 +1,6 @@
 /obj/item/ammo_magazine/hardpoint/tank_glauncher
 	name = "M92T Grenade Launcher Magazine"
-	desc = "A secondary armament grenade magazine."
+	desc = "A magazin loaded with M40 grenades. Used to reaload the Magazin feed M92T Grenade launcher."
```
Comment (harryob) [inline code/modules/vehicles/hardpoints/hardpoint_ammo/gl_ammo.dm:3]:
```suggestion
	desc = "A magazine loaded with M40 grenades. Used to reload the magazine fed M92T Grenade launcher."
```

File: code/modules/vehicles/hardpoints/hardpoint_ammo/minigun_ammo.dm [hunk 11]
```diff
@@ -1,7 +1,7 @@
 /obj/item/ammo_magazine/hardpoint/ltaaap_minigun
 	name = "LTAA-AP Minigun Magazine"
 	icon = 'icons/obj/items/weapons/guns/ammo_by_faction/uscm.dmi'
-	desc = "A primary armament minigun magazine."
+	desc = "A magazin of 7.62x51mm AP ammo for a heavy minigun. Filled to the brimm with highly precice armor-penetration rounds."
```
Comment (harryob) [inline code/modules/vehicles/hardpoints/hardpoint_ammo/minigun_ammo.dm:4]:
```suggestion
	desc = "A magazine of 7.62x51mm AP ammo for a heavy minigun. Filled to the brim with highly precise armor-penetrating rounds."
```

File: code/modules/vehicles/hardpoints/secondary/cupola.dm [hunk 12]
```diff
@@ -1,6 +1,6 @@
 /obj/item/hardpoint/secondary/m56cupola
 	name = "\improper M56 Cupola"
-	desc = "A secondary weapon for tanks that shoots bullets"
+	desc = "A secondary weapon for tanks. It's a M56D that was adjusted to be permanantly fixed to its mount. You swear you can still see some weld tacks."
```
Comment (harryob) [inline code/modules/vehicles/hardpoints/secondary/cupola.dm:3]:
```suggestion
	desc = "A secondary weapon for tanks. It's a M56D that was adjusted to be permanently fixed to its mount. You swear you can still see some weld tacks."
```

File: code/modules/vehicles/hardpoints/secondary/grenade_launcher.dm [hunk 13]
```diff
@@ -1,6 +1,6 @@
 /obj/item/hardpoint/secondary/grenade_launcher
 	name = "\improper M92T Grenade Launcher"
-	desc = "A secondary weapon for tanks that shoots grenades."
+	desc = "A magazin feed secondary grenade launcher for tanks that shoots M40 grenades."
```
Comment (harryob) [inline code/modules/vehicles/hardpoints/secondary/grenade_launcher.dm:3]:
```suggestion
	desc = "A magazine fed secondary grenade launcher for tanks that shoots M40 grenades."
```

File: code/modules/vehicles/hardpoints/wheels/treads.dm [hunk 14]
```diff
@@ -18,7 +18,7 @@
 
 /obj/item/hardpoint/locomotion/treads/robust
 	name = "\improper Reinforced Treads"
-	desc = "These treads are made of a tougher material and are more durable. However, the extra weight slows the tank down."
+	desc = "These treads are made of soldi steel plates and are more durable. However, the extra weight slows the tank down."
```
Comment (harryob) [inline code/modules/vehicles/hardpoints/wheels/treads.dm:21]:
```suggestion
	desc = "These treads are made of solid steel plates and are more durable. However, the extra weight slows the tank down."
```

**Review by harryob — CHANGES_REQUESTED** [review]

File: code/game/machinery/vending/vendor_types/crew/vehicle_crew.dm (hunk 5 above)
Comment (TheManWithNoHands) [reply code/game/machinery/vending/vendor_types/crew/vehicle_crew.dm:400]: It is a quint launcher, it 5 rocket loaded.

Comment (harryob) [reply code/game/machinery/vending/vendor_types/crew/vehicle_crew.dm:400]: huh. fair

File: code/modules/vehicles/hardpoints/hardpoint_ammo/autocannon_ammo.dm (hunk 9 above)
Comment (harryob) [reply code/modules/vehicles/hardpoints/hardpoint_ammo/autocannon_ammo.dm:3]: bump

**Pull request merged**